### PR TITLE
EN-58985: add focal for ruby 3.0.x

### PR DESCRIPTION
**Pull request by sarah-salinas**
## Before

runit-ruby-focal and ruby-focal have only ruby 3.1.x and 3.2.x options.

## After

runit-ruby-focal and ruby-focal also have an option for ruby 3.0.x.
This is what is used for frontend, storyteller and elections currently.

### CODE REVIEW

File: ruby-focal/3.0.x/Dockerfile
Comment (sarah-salinas) [inline ruby-focal/3.0.x/Dockerfile:11]: Copied from ruby-focal/3.1.x -- I only changed the ruby major version on lines 11 and line 41

File: runit-ruby-focal/3.0.x/Dockerfile
Comment (sarah-salinas) [inline runit-ruby-focal/3.0.x/Dockerfile:11]: Copied from runit-ruby-focal/3.1.x -- I only changed the ruby major version on lines 11 and 41.

File: ruby-focal/3.0.x/Dockerfile
Comment (sarah-salinas) [inline ruby-focal/3.0.x/Dockerfile:2]: I also changed this from the deprecated MAINTAINER to a LABEL, using the format here: https://github.com/socrata/shipyard/blob/main/base-focal/Dockerfile#L2

**Comment by dosieckityler** [comment]
What's the root need for a downgraded ruby in focal?  Are there specific language changes holding you back on the older version?  For which service/daemon is this impacting?  TIRe would love to help folks get onto newer versions of ruby.  There is a huge maintenance cost to multiple variants of base images such as this, so we'd like to avoid it if at all possible.

**Comment by sarah-salinas** [comment]
I know that Helena ran into issues when she tried to upgrade to ruby 3.1.x for storyteller and frontend that were non-trivial.  @hiddensanctum, I know you worked with Helena on the ruby upgrade, do you have enough insight into the issues so that you can give a general LOE for doing that work?

**Comment by hiddensanctum** [comment]
> I know that Helena ran into issues when she tried to upgrade to ruby 3.1.x for storyteller and frontend that were non-trivial. @hiddensanctum, I know you worked with Helena on the ruby upgrade, do you have enough insight into the issues so that you can give a general LOE for doing that work?

I believe the issue we ran into was Mac OSX specific. Back in Feb there was a bug where [Macs couldn't build native ruby gems](https://bugs.ruby-lang.org/issues/19403). It looks like that might have been fixed, but I also would need something with a mac to try

**Comment by sarah-salinas** [comment]
@thehouseplant does that mean that I can merge this?

I looked into upgrading to ruby 3.1.x and there were a number of gems that will need to be addressed including zookeeper . . . which I believe takes some coordination.

**Comment by sarah-salinas** [comment]
To clarify, I agree that we should move forward with converting storyteller, frontend and elections to 3.2.x, and I would be happy to pair with someone to learn how to do it, but I do not think that we will be able to do it by the due date on the ticket.   I am also concerned that it will involve updating dependencies that we manage like socrata-eurbates and I'm not sure if anyone is available to do that soon.

My understanding is that since frontend and storyteller are part of the shared release process, we need to get the ubuntu change merged by Wednesday for it to get to fedramp.  I'm not sure how hard or flexible the deadline is.

**Comment by dosieckityler** [comment]
If we're hard blocked on getting services onto 3.1.x or 3.2.x, the migration from bionic to focal does take priority.  I can give a :shipit: on the PR, but then TIRe will need to spawn some work to get these new base images created and integrated into all our tooling.

**Comment by sarah-salinas** [comment]
Okay, I'll merge and then you can let me know when it'll be ready to use.
I made a ticket for upgrading to ruby 3.2.x sooner than later and will discuss with my team about getting help with it: https://socrata.atlassian.net/browse/EN-59084

**Review by dosieckityler — APPROVED** [review]

**Pull request merged**
